lib/view_component/base.rb

```diff
@@ -180,8 +184,20 @@ def compile(raise_template_errors: false)
           return false
         end
 
+        # If template name annotations are turned on, a line is dynamically
+        # added with a comment. In this case, we want to return a different
+        # starting line number so errors that are raised will point to the
+        # correct line in the component template.
+        line_number =
+          if ActionView::Base.respond_to?(:annotate_template_file_names) &&
+            ActionView::Base.annotate_template_file_names
+            -2
```

Comment:
Would it be worth adding a regression test for this case since we already have one for `-1`?

Reply:
Our test will cover both cases, since we run both Rails 6.1 and previous versions in our matrix builds ❤️